base/version_comparable_test.go

```diff
@@ -115,6 +115,71 @@ func TestInvalidComparableVersion(t *testing.T) {
 	}
 }
 
+func TestJSONRoundTrip(t *testing.T) {
+	json, err := JSONMarshal(ProductVersion)
+	require.NoError(t, err)
+	var version ComparableVersion
+	err = JSONUnmarshal(json, &version)
+	require.NoError(t, err)
+	require.True(t, ProductVersion.Equal(&version))
+	require.Equal(t, ProductVersion.String(), version.String())
+}
+
+func TestAtLeastMinorDowngradeVersion(t *testing.T) {
+	testCases := []struct {
+		versionA       string
+		versionB       string
+		minorDowngrade bool
+	}{
+		{
+			versionA:       "1.0.0",
+			versionB:       "1.0.0",
+			minorDowngrade: false,
```

Comment:
I think versionA: `1.0.0.2` versionB: `1.0.0.1` would be a good case to add to be sure we avoid regressions.